413 error pages

davidtrussler opened the pull request:
[Trello](https://trello.com/c/7vZhUzMI/413-error-pages-should-match-the-design-system-for-things-like-page-not-found-404-and-other-server-errors)

The changes here add custom error pages:
- Adds a new errors controller to catch errors and render the appropriate view
- Updates the routes file to route errors via the appropriate controller actions
- Updates the application config file to serve error pages from the app instead of the default static pages
- Adds new views for five categories of error page
- Deletes the default static pages
- Adds tests for displaying the new error pages

Screenshots below.

|Page|Screenshot|
|-|-|
|400 - Bad request|![Screenshot 2023-09-27 at 12 38 23](https://github.com/alphagov/whitehall/assets/6080548/a1bf9a2a-06ea-4fcd-885e-bdb76b1646bb)|
|403 - Forbidden|![Screenshot 2023-09-27 at 12 40 26](https://github.com/alphagov/whitehall/assets/6080548/fe9416bf-a407-4b17-8cfe-f56e5a17eefa)|
|404 - Not found|![Screenshot 2023-09-27 at 12 41 18](https://github.com/alphagov/whitehall/assets/6080548/92194a52-389a-48c1-896a-089d3c958f87)|
|422 - Unprocessable entity|![Screenshot 2023-09-27 at 12 41 58](https://github.com/alphagov/whitehall/assets/6080548/a7910957-3da6-47fe-a3d9-c7497f002ede)|
|500 - Internal server error|![Screenshot 2023-09-27 at 12 42 36](https://github.com/alphagov/whitehall/assets/6080548/bddc3164-6428-44a4-b06b-e77e990ac8d4)|

nnagewad commented:
Screenshots look great @davidtrussler

nnagewad commented:
@davidtrussler and I went over the spacing. It has been updated and looks great

nnagewad commented:
@davidtrussler. Rosa and I have updated the messaging for the pages. Image below reflect the content changes as it sounds more like GOV.UK and there are ways for users to resolve the issue at hand.

![Error pages](https://github.com/alphagov/whitehall/assets/8485154/445a11ad-5d4f-47d4-bcd2-f659d72d554f)

davidtrussler commented:
> @davidtrussler. Rosa and I have updated the messaging for the pages. Image below reflect the content changes as it sounds more like GOV.UK and there are ways for users to resolve the issue at hand.
> 
> For list items, I actually used the list component: https://components.publishing.service.gov.uk/component-guide/list

@nnagewad @rosaryou I've updated the screenshots here to reflect the design and content changes. If you're OK with them I'll move the card back to "In Progress" as there is still some work to do prior to a technical review.

nnagewad commented:
@davidtrussler great job bud 👍🏽

ollietreend commented:
> **There is a mistake in the URL**

Any reason why we're not going with the more standard "Page not found" wording for the 404 error page?

I think that's usually what people expect to see. And that'd make it consistent with the GOV.UK 404 error page: https://www.gov.uk/404

![Screenshot of the GOV.UK 404 error page](https://github.com/alphagov/whitehall/assets/7735945/5ecd2d5d-1618-4851-946c-df348b774d44)

cc @nnagewad @rosaryou

rosaryou commented:
Because we thought the error message applied to incorrect URLs but if this
is not the case, we can have another look

On Thu, 28 Sept 2023 at 14:45, Ollie Treend ***@***.***>
wrote:

> *There is a mistake in the URL*
>
> Any reason why we're not going with the more standard "Page not found"
> wording for the 404 error page?
>
> I think that's usually what people expect to see. And that'd make it
> consistent with the GOV.UK 404 error page: https://www.gov.uk/404
>
> [image: Screenshot of the GOV.UK 404 error page]
> <https://user-images.githubusercontent.com/7735945/271304285-5ecd2d5d-1618-4851-946c-df348b774d44.png>
>
> cc @nnagewad <https://github.com/nnagewad> @rosaryou
> <https://github.com/rosaryou>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alphagov/whitehall/pull/8289#issuecomment-1739231818>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALPZFTISQSOTD45RCCFQ4PTX4V5PRANCNFSM6AAAAAA5BXTW6M>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Rosa Ryou*
Content learning designer
GOV.UK Publishing Experience
Government Digital Services, Cabinet Office
M: 07 825 657 250

farahTW commented on test/functional/admin/errors_controller_test.rb:
Hey David 
ErrorsControllerTest should inherit from.  ActionController::TestCase I believe not Integration.
Rest looks good to me.

davidtrussler replied:
After a discussion with @farahTW we will leave this as it is for now, though it might want some attention in future.

farahTW reviewed (approved):
LGTM
